docs/src/DeveloperDocumentation/CachedFunctions.md

```diff
@@ -0,0 +1,367 @@
+> - `fqPolyRepField(f::fpPolyRingElem, s::Symbol, cached::Bool = true; check::Bool=true)`
+> - `FqField(char::ZZRingElem, deg::Int, s::Symbol, cached::Bool = true)`
+> - `FqField(f::ZZModPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqField(f::FpPolyRingElem, s::Symbol, cached::Bool = true;  check::Bool = true)`
+> - `FqField(f::zzModPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqField(f::fpPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(char::ZZRingElem, deg::Int, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(f::ZZModPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(f::FpPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(char::ZZRingElem, deg::Int, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(f::ZZModPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `FqPolyRepField(f::FpPolyRingElem, s::Symbol, cached::Bool = true; check::Bool = true)`
+> - `fqPolyRepMPolyRing(R::fqPolyRepField, s::Vector{Symbol}, S::Symbol, cached::Bool = true)`
+> - `PadicField(p::ZZRingElem, prec::Int; cached::Bool = true, check::Bool = true)`
+> - `QadicField(p::ZZRingElem, d::Int, prec::Int, var::String = "a"; cached::Bool = true, check::Bool = true)`
+> - `ZZRelPowerSeriesRing(prec::Int, s::Symbol, cached::Bool = true)`
+> - `ZZAbsPowerSeriesRing(prec::Int, s::Symbol, cached::Bool = true)`
+> - `FlintPuiseuxSeriesRing{T}(R::Ring, cached::Bool = true)`
+> - `FlintPuiseuxSeriesField{T}(R::Field, cached::Bool = true)`
+> - `ZZLaurentSeriesRing(prec::Int, s::Symbol, cached::Bool = true)`
+> - `QQRelPowerSeriesRing(prec::Int, s::Symbol, cached::Bool = true)`
+> - `QQAbsPowerSeriesRing(prec::Int, s::Symbol, cached::Bool = true)`
+> - `FqPolyRing(R::FqField, s::Symbol, cached::Bool = true)`
+> - `FqPolyRepPolyRing(R::FqPolyRepField, s::Symbol, cached::Bool = true)`
+> - `fqPolyRepPolyRing(R::fqPolyRepField, s::Symbol, cached::Bool = true)`
+> - `FqMPolyRing(R::FqField, s::Vector{Symbol}, internal_ordering::Symbol = :lex, cached::Bool = true)`
+> - `similar(f::PolyRingElem, R::QQField, s::Symbol=var(parent(f)`
+> - `matrix_space(R::ZZRing, r::Int, c::Int; cached::Bool = true)`
+> - `matrix_space(R::ZZModRing, r::Int, c::Int; cached::Bool = true)`
+> - `similar(f::PolyRingElem, R::ZZModRing, s::Symbol=var(parent(f)`
+> - `similar(f::PolyRingElem, R::ZZRing, s::Symbol=var(parent(f)`
+> - `matrix_space(R::FqField, r::Int, c::Int; cached::Bool = true)`
+> - `similar(f::PolyRingElem, R::FqField, s::Symbol=var(parent(f)`
+> - `matrix_space(R::FqPolyRepField, r::Int, c::Int; cached::Bool = true)`
+> - `matrix_space(R::fqPolyRepField, r::Int, c::Int; cached::Bool = true)`
+> - `similar(f::PolyRingElem, R::fqPolyRepField, s::Symbol=var(parent(f)`
+> - `similar(f::PolyRingElem, R::FqPolyRepField, s::Symbol=var(parent(f)`
```

Comment:
@aaruni96 your magical pipe seems to cut off some function signatures like here. (Just saying. As long as we don't know what we *do* with this list, there is probably no point in debugging this.)

Reply:
I see, it cuts off at the first closing parenthesis. Its probably worth fixing, as it might be confusing later as to why something without `cached` seems to appear in the list.

I'll change the `sed` pattern.